hazelcast/test/src/compact_test.cpp

```diff
@@ -634,6 +650,351 @@ struct hz_serializer<compact::test::employee_dto> : public compact_serializer
+            {42                  , 42   , 0                   },
+            {42                  , -42  , -1212835703325587522},
+            {0                   , 0    , 0                   },
+            {-123456789          , 0    , 7049212178818848951 },
+            {123456789           , 127  , -8322440716502314713},
+            {127                 , -128 , -7333697815154264656},
+        }
+    );
+}
+
+TEST_F(CompactRabinFingerprintTest, test_i32_fingerprint)
+{
+    check_each(
+        std::vector<entry_t<int>>
+        { // Before               Val            After(Expected)
+            {int64_min           , 2147483647  , 6066553457199370002 },
+            {9223372036854775807 , int32_min   , 6066553459773452525 },
+            {9223372036854707    , 42          , -961937498224213201 },
+            {-42                 , -42         , 4294967295          },
+            {42                  , 42          , 0                   },
+            {42                  , -442        , 7797744281030715531 },
+            {0                   , 0           , 0                   },
+            {-123456789          , 0           , -565582369564281851 },
+            {123456786669        , 42127       , 7157681543413310373 },
+            {2147483647          , int32_min    , -7679311364898232185}
+        }
+    );
+}
+
+TEST_F(CompactRabinFingerprintTest, test_str_fingerprint)
+{
+    check_each(
+        std::vector<entry_t<std::string>>
+        {
+            {0                        , "hazelcast"  , 8164249978089638648  },
+            {-31231241235             , "üğişçö"     , 6128923854942458838 },
+            {41231542121235           , "😀 😃 😄" , -6875080751809013377 },
+            {rabin_finger_print::INIT , "STUdent"    , 1896492170246289820  },
+            {rabin_finger_print::INIT , "aü😄"      , -2084249746924383631 },
+            {rabin_finger_print::INIT , ""           , -2316162475121075004 },
+            {-123321                  , "xyz"        , 2601391163390439688  },
+            {132132123132132          , "    ç"      , -7699875372487088773 },
+            {42                       , "42"         , 7764866287864698590  },
+            {-42                      , "-42"        , -3434092993477103253 }
+        }
+    );
+}
+
+TEST_F(CompactRabinFingerprintTest, test_schema)
+{
+    using hazelcast::client::serialization::compact_writer;
+    using hazelcast::client::serialization::pimpl::schema_writer;
+    using hazelcast::client::serialization::pimpl::field_kind;
+
+    schema_writer  s_writer { "SomeType" };
+    auto writer = hazelcast::client::serialization::pimpl::create_compact_writer(&s_writer);
+
+    writer.write_int32("id", 0);
+    writer.write_string("name" , std::string {} );
+    writer.write_int8("age" , 0);
+    writer.write_array_of_timestamp("times", boost::none);
+
+    auto schema_id = std::move(s_writer).build().schema_id();
+    ASSERT_EQ(-5445839760245891300, schema_id);
```

Comment:
what might be the reason to have different assert values from Java API. As I see, the writer is composed from the same parameters but in Java Client side the output is compared with 3662264393229655598L

Reply:
Enum values are not correct for this PR. It will be fixed in the [PR](https://github.com/hazelcast/hazelcast-cpp-client/pull/1039) when this PR is merged to master.